third_party/move/move-vm/runtime/src/storage/ty_depth_checker.rs

```diff
@@ -175,19 +176,6 @@ where
                 let formula = visit_struct!(idx);
                 check_depth!(formula.solve(&ty_arg_depths))
             },
```

Comment:
If we remove this, we should update the documentation, which says "Checks depths for instantiated types.". It doesn't say "check depth of value which can be constructed by type". Or for which is this actually used? If its used for both value depth and for binding type depth, we should leave it indeed.

If we remove it there needs to be some comment why functions are now treated as '1', it doesn't make sense if you look locally at this code.